### PR TITLE
Add FindDrake.cmake so we can write c++ code that targets drake

### PR DESCRIPTION
**Pull request by manuelli**
This is a temporary fix for the issue in #24. Also relevant to https://github.com/RobotLocomotion/drake/issues/5456.

The current solution is to use a custom ``FindDrake.cmake`` and ``FindEigen.cmake`` to actually find ``Drake`` and ``Eigen`` rather than relying on ``drake-config.cmake`` which is broken. 

@patmarion @siyuanfeng-tri @EricCousineau-TRI @sammy-tri

### CODE REVIEW

File: cmake/modules/FindDrake.cmake [hunk 1]
```diff
@@ -0,0 +1,67 @@
+if(DRAKE_DIR)
+  set(_include_dir_hint ${DRAKE_DIR}/include)
+  set(_lib_dir_hint ${DRAKE_DIR}/lib)
+  message("DRAKE_DIR is ${DRAKE_DIR}")
+endif()
+
+message("Lucas message")
```
Comment (EricCousineau-TRI) [inline cmake/modules/FindDrake.cmake]: nit Leftover debug message?

**Comment by EricCousineau-TRI** [comment]
With this, would it merit removing the `WITH_ISSUE_5456_WORKAROUND` hack?

File: cmake/modules/FindDrake.cmake (hunk 1 above)
Comment (manuelli) [inline cmake/modules/FindDrake.cmake]: whoops, just fixed.

**Comment by manuelli** [comment]
Yeah probably makes sense to remove the `WITH_ISSUE_5456_WORKAROUND` hack. I'll update my branch.

**Comment by patmarion** [comment]
LGTM.  Maybe you could remove some of the debug print messages in the cmake code.  Also, you should remove the README instructions that reference `WITH_ISSUE_5456_WORKAROUND`.

It could also be improved so that the user code doesn't need to worry about `pthread` and `yaml-cpp`, but we can fix that up in a future PR, this is at least much better than it was.

**Comment by EricCousineau-TRI** [comment]
Did a quick `grep` on the workaround: Would it also make sense to remove the `Common Build Errors` blurb in `README.md`?

**Comment by manuelli** [comment]
Thanks for the comments guys. Since we don't have CI yet if you have time and want to pull this branch down and build justt to make sure it's working on different platforms that would be great.

**Comment by manuelli** [comment]
I just realized I haven't update the ObjectDetection CMakeLists.txt yet so hold off on merging.

**Comment by EricCousineau-TRI** [comment]
Yup, just set it to build on my system, and it seems like GMock is the only thing amiss here:
```
[ 52%] Performing configure step for 'ObjectDetection'
loading initial cache file .../spartan/build/tmp/ObjectDetection-cache-Release.cmake
Eigen3 include dir.../spartan/build/install/include/eigen3
CMake Error at /usr/share/cmake-3.5/Modules/CMakeFindDependencyMacro.cmake:65 (find_package):
  By not providing "FindGMock.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "GMock", but
  CMake did not find one.

  Could not find a package configuration file provided by "GMock" with any of
  the following names:

    GMockConfig.cmake
    gmock-config.cmake

  Add the installation prefix of "GMock" to CMAKE_PREFIX_PATH or set
  "GMock_DIR" to a directory containing one of the above files.  If "GMock"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  .../spartan/build/install/lib/drake/cmake/drake-config.cmake:40 (find_dependency)
  src/CMakeLists.txt:14 (find_package)
```

Just post when you have it updated, and I'll try again.

**Comment by patmarion** [comment]
That looks like it is calling the broken `drake-config.cmake`, see here:

```
Call Stack (most recent call first):
  .../spartan/build/install/lib/drake/cmake/drake-config.cmake:40 (find_dependency)
  src/CMakeLists.txt:14 (find_package)
```

**Comment by manuelli** [comment]
Yeah I am just going to disable ObjectDetection for now. @gizatt can fix it when he has time. I am still making a few fixes on my end. Will keep pushing updates and let you know when I am happy.

**Comment by manuelli** [comment]
Isn't `if(WITH_DRIVERS)` in the next `if` statement?

**Comment by patmarion** [comment]
sorry, misread.  youre right

**Comment by manuelli** [comment]
I think that this is ready for merge now. I have built it on a couple of machines so far and it has been working. I also set `USE_SYTEM_VTK` to be `OFF` by default. It inevitably leads to build issues otherwise if you have the wrong version of vtk. Better to default to the safe (but slower) option I think.

@patmarion merge if you are happy.

**Comment by manuelli** [comment]
These changes are included in an updated form in #105.

**Pull request closed without merging**